mono/mini/mini-darwin.c

```diff
@@ -79,6 +79,29 @@ mono_runtime_install_handlers (void)
 {
 	mono_runtime_posix_install_handlers ();
 
+#if !defined (HOST_WATCHOS) && !defined (HOST_TVOS)
+	/* LLDB installs task-wide Mach exception handlers. XNU dispatches Mach
+	 * exceptions first to any registered "activation" handler and then to
+	 * any registered task handler before dispatching the exception to a
+	 * host-wide Mach exception handler that does translation to POSIX
+	 * signals. This makes it impossible to use LLDB with an
+	 * implicit-null-check-enabled Mono; continuing execution after LLDB
+	 * traps an EXC_BAD_ACCESS will result in LLDB's EXC_BAD_ACCESS handler
+	 * being invoked again. This also interferes with the translation of
+	 * SIGFPEs to .NET-level ArithmeticExceptions. Work around this here by
+	 * installing a no-op task-wide Mach exception handler for
+	 * EXC_BAD_ACCESS and EXC_ARITHMETIC.
+	 */
+	kern_return_t kr = task_set_exception_ports (
```

Comment:
In Windbg when you stop on an exception, you can g or gh (go handled) or gn (go not handled), where g is short for one of these. So when you have code for example, that catches a breakpoint exception, you say gn, to let that code see it. Instead of gh which would quash it.

I wonder if lldb has similar and that would help.

Does this workaround mean null dereferences do not by default break in to debugger?
That would seem usually unfortunate, since they are usually bugs you want to see.

Reply:
> Does this workaround mean null dereferences do not by default break in to debugger?
That would seem usually unfortunate, since they are usually bugs you want to see.

`lldb` will still see `SIGSEGV` and is able to continue execution from that (i.e. let the debugee handle `SIGSEGV`).